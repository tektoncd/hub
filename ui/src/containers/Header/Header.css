.hub-header-login {
  color: white;
  font-size: 1em;
}

.header-search-hint svg {
  vertical-align: 0em !important;
  margin-left: 0.2em;
  cursor: pointer;
}
