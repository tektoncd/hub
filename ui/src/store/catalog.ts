import { Instance, types } from 'mobx-state-tree';
import { Icons } from '../common/icons';

const icons: { [catalog: string]: Icons } = {
  official: Icons.Cat,
  verified: Icons.Certificate,
  community: Icons.User
};

export const Catalog = types
  .model({
    id: types.identifierNumber,
    name: types.optional(types.string, ''),
    type: types.optional(types.string, ''),
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }))
  .views((self) => ({
    get icon(): Icons {
      return icons[self.type] || Icons.Unknown;
    }
  }));

export type ICatalog = Instance<typeof Catalog>;
export type ICatalogStore = Instance<typeof CatalogStore>;

export const CatalogStore = types
  .model({
    items: types.map(Catalog)
  })

  .actions((self) => ({
    add(item: ICatalog) {
      self.items.put({ id: item.id, name: item.name, type: item.type });
    },
    clearSelected() {
      self.items.forEach((c) => {
        c.selected = false;
      });
    }
  }))

  .views((self) => ({
    get values() {
      return Array.from(self.items.values());
    },

    get selected() {
      const list = new Set();
      self.items.forEach((c: ICatalog) => {
        if (c.selected) {
          list.add(c.id);
        }
      });

      return list;
    }
  }));
