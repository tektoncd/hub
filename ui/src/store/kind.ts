import { types, Instance } from 'mobx-state-tree';
import { Icons } from '../common/icons';

const icons: { [kind: string]: Icons } = {
  Task: Icons.Build,
  Pipeline: Icons.Domain
};

export const Kind = types
  .model({
    name: types.identifier,
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }))
  .views((self) => ({
    get icon(): Icons {
      return icons[self.name] || Icons.Unknown;
    }
  }));

export type IKind = Instance<typeof Kind>;
export type IKindStore = Instance<typeof KindStore>;

export const KindStore = types
  .model({
    items: types.optional(types.map(Kind), {})
  })

  .actions((self) => ({
    add(item: string) {
      self.items.put({ name: item, selected: false });
    },

    clearSelected() {
      self.items.forEach((k) => {
        k.selected = false;
      });
    }
  }))

  .views((self) => ({
    get values() {
      return Array.from(self.items.values());
    },

    get selected() {
      const list = new Set();
      self.items.forEach((c: IKind) => {
        if (c.selected) {
          list.add(c.name);
        }
      });

      return list;
    }
  }));
