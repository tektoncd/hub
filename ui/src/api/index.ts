import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

interface Token {
  token: string;
  refreshInterval: string;
  expiresAt: number;
}

interface TokenData {
  access: Token;
  refresh: Token;
}

export interface AuthResponse {
  data: TokenData;
}

export interface Rating {
  rating: number;
}

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
  resourceVersion(resourceId: number): Promise<IVersion>;
  versionUpdate(versionId: number): Promise<IVersion>;
  authentication(authCode: string): Promise<AuthResponse>;
  readme(rawURL: string): Promise<string>;
  yaml(rawURL: string): Promise<string>;
  getRating(resourceId: number, token: string): Promise<Rating>;
  setRating(resourceId: number, token: string, rating: number): Promise<void | null>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async authentication(authCode: string) {
    try {
      return axios
        .post(`${API_URL}/auth/login?code=${authCode}`)
        .then((response) => response.data)
        .catch((err) => Promise.reject(err.response));
    } catch (err) {
      return err;
    }
  }

  async resourceVersion(resourceId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async versionUpdate(versionId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/version/${versionId}`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async readme(rawURL: string) {
    try {
      const URL = rawURL.substring(0, rawURL.lastIndexOf('/') + 1);
      return axios.get(`${URL}/README.md`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async yaml(rawURL: string) {
    try {
      const newLine = '\n';
      return axios.get(`${rawURL}`).then((response) => '```yaml' + newLine + response.data);
    } catch (err) {
      return err.response;
    }
  }

  async getRating(resourceId: number, token: string) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/rating`, {
          headers: {
            Authorization: `Bearer ${token}`
          }
        })
        .then((response) => response.data)
        .catch((err) => Promise.reject(err.response));
    } catch (err) {
      return err;
    }
  }

  async setRating(resourceId: number, token: string, rating: number) {
    try {
      return axios
        .put(
          `${API_URL}/resource/${resourceId}/rating`,
          { rating: rating },
          {
            headers: {
              Authorization: `Bearer ${token}`
            }
          }
        )
        .then((response) => response.data)
        .catch((err) => Promise.reject(err.response));
    } catch (err) {
      return err;
    }
  }
}
