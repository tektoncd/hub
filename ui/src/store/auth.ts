import { flow, getEnv, Instance, types } from 'mobx-state-tree';
import { Api } from '../api';
import { titleCase } from '../common/titlecase';

export const TokenInfo = types.model('TokenInfo', {
  token: types.optional(types.string, ''),
  expiresAt: types.optional(types.number, 0),
  refreshInterval: types.optional(types.string, '')
});

export const Error = types.model('Error', {
  status: types.optional(types.number, 0),
  customMessage: types.optional(types.string, ''),
  serverMessage: types.optional(types.string, '')
});

export type ITokenInfo = Instance<typeof TokenInfo>;
export type IError = Instance<typeof Error>;

export interface AuthCodeProps {
  code: string;
}

interface Refresh {
  (): void;
}

export const AuthStore = types
  .model('AuthStore', {
    accessTokenInfo: types.optional(TokenInfo, {}),
    refreshTokenInfo: types.optional(TokenInfo, {}),
    isLoading: true,
    isAuthenticated: false,
    userRating: types.optional(types.number, 0),
    authErr: types.optional(Error, {}),
    ratingErr: types.optional(Error, {})
  })
  .actions((self) => ({
    addAccessTokenInfo(item: ITokenInfo) {
      self.accessTokenInfo.token = item.token;
      self.accessTokenInfo.expiresAt = item.expiresAt;
      self.accessTokenInfo.refreshInterval = item.refreshInterval;
    },
    addRefreshTokenInfo(item: ITokenInfo) {
      self.refreshTokenInfo.token = item.token;
      self.refreshTokenInfo.expiresAt = item.expiresAt;
      self.refreshTokenInfo.refreshInterval = item.refreshInterval;
    },
    setIsAuthenticated(l: boolean) {
      self.isAuthenticated = l;
    },
    setUserRating(rating: number) {
      self.userRating = rating;
    },
    setLoading(l: boolean) {
      self.isLoading = l;
    },
    setErrorMessage(error: IError) {
      switch (error.status) {
        case 400:
          error.customMessage = 'Bad Request';
          break;
        case 401:
          error.customMessage = 'Unauthorized. Please login again!';
          break;
        case 404:
          error.customMessage = 'Requested resource not found';
          break;
        case 500:
          error.customMessage = 'Internal server error. Please try again after some time!';
          break;
        default:
          error.customMessage = '';
      }
    },
    onFailure(err: Error) {
      self.authErr.serverMessage = err.toString();
      self.authErr.status = 400;
      self.authErr.customMessage = 'GitHub login failed';
    },
    logout() {
      localStorage.clear();
      self.isAuthenticated = false;
      self.isLoading = false;
      self.userRating = 0;
    }
  }))
  .views((self) => ({
    get api(): Api {
      return getEnv(self).api;
    }
  }))
  .actions((self) => ({
    authenticate: flow(function* (authCode: AuthCodeProps, refresh?: Refresh) {
      try {
        self.setLoading(true);

        const { api } = self;
        const json = yield api.authentication(authCode.code);

        const userDetails = json.data;
        self.addAccessTokenInfo(userDetails.access);
        self.addRefreshTokenInfo(userDetails.refresh);

        self.setIsAuthenticated(true);

        if (refresh) {
          refresh();
        }
      } catch (err) {
        const error: IError = {
          status: err.status,
          serverMessage: titleCase(err.data.message),
          customMessage: ''
        };
        self.setErrorMessage(error);
        self.authErr = error;
      }
      self.setLoading(false);
    }),

    getRating: flow(function* (resourceId: number) {
      try {
        self.setLoading(true);
        const { api } = self;

        if (self.isAuthenticated) {
          const json = yield api.getRating(resourceId, self.accessTokenInfo.token);
          self.setUserRating(json.rating);
        }
      } catch (err) {
        const error: IError = {
          status: err.status,
          serverMessage: titleCase(err.data.message),
          customMessage: ''
        };
        self.setErrorMessage(error);
        self.ratingErr = error;
      }
      self.setLoading(false);
    }),

    setRating: flow(function* (resourceId: number, rating: number) {
      try {
        self.setLoading(true);

        const { api } = self;

        if (self.isAuthenticated) {
          yield api.setRating(resourceId, self.accessTokenInfo.token, rating);
          self.setUserRating(rating);
        }
      } catch (err) {
        const error: IError = {
          status: err.status,
          serverMessage: titleCase(err.data.message),
          customMessage: ''
        };
        self.setErrorMessage(error);
        self.ratingErr = error;
      }
      self.setLoading(false);
    })
  }));
export type IAuthStore = Instance<typeof AuthStore>;
