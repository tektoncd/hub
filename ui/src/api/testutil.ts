import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;
    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async resources() {
    const data = `${this.dataDir}/resources.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IResource>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async resourceVersion(resourceId: number) {
    const data = `${this.dataDir}/resource_versions_${resourceId}.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IVersion>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async versionUpdate(versionId: number) {
    const data = `${this.dataDir}/version_details_${versionId}.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IVersion>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
