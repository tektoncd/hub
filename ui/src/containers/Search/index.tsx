import React from 'react';
import { useNavigate } from 'react-router-dom';
import { useHotkeys } from 'react-hotkeys-hook';
import { useMst } from '../../store/root';
import { observer } from 'mobx-react';
import { TextInput } from '@patternfly/react-core';
import { useDebouncedCallback } from 'use-debounce';
import { assert } from '../../store/utils';
import './Search.css';

export const TagsKeyword = 'tags:';

const Search: React.FC = observer(() => {
  const { resources } = useMst();
  const [value, setValue] = React.useState(resources.search);

  interface SearchInfo {
    search: string;
    tags: Array<string>;
  }

  React.useEffect(() => {
    setValue(resources.search);
  }, [resources.search]);

  const inputRef = React.useRef<HTMLInputElement>(null);
  useHotkeys('/', () => {
    // Without timeout the hotkey will be sent to select input
    setTimeout(() => {
      assert(inputRef.current);
      inputRef.current.focus();
    }, 20);
  });

  /* Debounce callback
     This function delay the invoking func for 500ms after
     when users stop searching
  */
  const debounced = useDebouncedCallback(
    (data: SearchInfo) => {
      resources.setSearchedTags(data.tags);
      resources.setSearch(data.search);
    },
    // Delay in ms
    500
  );

  const onSearchChange = () => {
    const currentInput = inputRef.current;
    assert(currentInput);
    setValue(currentInput.value);

    const searchInput = currentInput.value;
    let tagList: Array<string>;

    if (searchInput.includes(TagsKeyword)) {
      const tagsString = searchInput.substr(5); // Removes `tags:` from search Input
      tagList = tagsString
        .split(',')
        .filter((item: string) => item !== '')
        .map((item: string) => item.trim());
    } else {
      tagList = [];
    }

    const searchedData: SearchInfo = {
      search: searchInput,
      tags: tagList
    };

    debounced(searchedData);
  };

  const history = useNavigate();
  const onSearchKeyPress = (e: React.KeyboardEvent<HTMLElement>) => {
    if (e.key === 'Enter') {
      e.preventDefault();
      if (window.location.pathname !== '/') history('/');
    }
    return;
  };

  return (
    <TextInput
      ref={inputRef}
      type="search"
      value={value}
      onChange={onSearchChange}
      onKeyPress={onSearchKeyPress}
      aria-label="text input example"
      placeholder="Search for resources..."
      spellCheck="false"
      className="hub-search"
    />
  );
});

export default Search;
