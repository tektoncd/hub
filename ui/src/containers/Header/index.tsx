import React from 'react';
import { Link } from 'react-router-dom';
import { observer } from 'mobx-react';
import { useHistory } from 'react-router-dom';
import {
  PageHeader,
  Brand,
  PageHeaderTools,
  Text,
  TextVariants,
  GridItem,
  Grid,
  Modal,
  ModalVariant,
  TextContent,
  TextList,
  TextListItem
} from '@patternfly/react-core';
import logo from '../../assets/logo/logo.png';
import { IconSize } from '@patternfly/react-icons';
import Search from '../../containers/Search';
import UserProfile from '../UserProfile';
import { useMst } from '../../store/root';
import './Header.css';
import { scrollToTop } from '../../common/scrollToTop';
import Icon from '../../components/Icon';
import { Icons } from '../../common/icons';

const Header: React.FC = observer(() => {
  const { user } = useMst();
  const history = useHistory();
  const [isModalOpen, setIsModalOpen] = React.useState(false);

  const headerTools = (
    <PageHeaderTools>
      <Grid>
        <GridItem span={10}>
          <Search />
        </GridItem>
        <GridItem span={1} onClick={() => setIsModalOpen(true)} className="header-search-hint">
          <Icon id={Icons.Help} size={IconSize.sm} label={'search-tips'} />
        </GridItem>
      </Grid>
      {user.isAuthenticated && user.refreshTokenInfo.expiresAt * 1000 > global.Date.now() ? (
        <UserProfile />
      ) : (
        <Text component={TextVariants.h3}>
          <Link to="/login" style={{ textDecoration: 'none' }}>
            <span className="hub-header-login">Login</span>
          </Link>
        </Text>
      )}
    </PageHeaderTools>
  );

  const homePage = () => {
    if (!window.location.search) history.push('/');
    scrollToTop();
  };

  return (
    <React.Fragment>
      <PageHeader
        logo={<Brand src={logo} alt="Tekton Hub Logo" onClick={homePage} />}
        headerTools={headerTools}
      />
      <Modal
        variant={ModalVariant.small}
        title="Search tips:"
        isOpen={isModalOpen}
        onClose={() => setIsModalOpen(false)}
      >
        <Grid>
          <TextContent>
            <TextList>
              <TextListItem>Press `/` to quickly focus on search.</TextListItem>
              <TextListItem>Search resources by name, displayName, and tags.</TextListItem>
              <TextListItem>
                Filter resources by tags using the qualifier like `tags:tagA,tagB`
              </TextListItem>
            </TextList>
          </TextContent>
        </Grid>
      </Modal>
    </React.Fragment>
  );
});
export default Header;
