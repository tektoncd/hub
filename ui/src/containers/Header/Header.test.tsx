import React from 'react';
import { mount } from 'enzyme';
import { when } from 'mobx';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '.';
import Search from '../../containers/Search';
import UserProfile from '../UserProfile';
import { FakeHub } from '../../api/testutil';
import { createProviderAndStore } from '../../store/root';

const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);
const { Provider, root } = createProviderAndStore(api);

describe('Header', () => {
  it('should render the header component and find Search component', () => {
    const component = mount(
      <Provider>
        <Router>
          <Header />
        </Router>
      </Provider>
    );

    expect(component.find(Search).length).toBe(1);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should find Login', () => {
    const component = mount(
      <Provider>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    expect(component.find('span').text()).toBe('Login');
  });

  it('should render user profile', (done) => {
    const component = mount(
      <Provider>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    const { user } = root;
    const code = {
      code: 'foo'
    };
    user.authenticate(code);
    when(
      () => {
        return !user.isLoading;
      },
      () => {
        setTimeout(() => {
          expect(user.isAuthenticated).toBe(true);
          component.update();
          expect(component.find(UserProfile).length).toEqual(1);
          done();
        }, 0);
      }
    );
  });
});
