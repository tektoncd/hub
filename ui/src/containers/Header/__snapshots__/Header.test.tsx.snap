// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the header component and find Search component 1`] = `
"<Provider>
   
  <BrowserRouter>
    <Router basename={[undefined]} location={{...}} navigationType=\\"POP\\" navigator={{...}}>
      <Memo(observerComponent)>
        <PageHeader logo={{...}} headerTools={{...}}>
          <header role={[undefined]} className=\\"pf-c-page__header\\">
            <div className=\\"pf-c-page__header-brand\\">
              <a className=\\"pf-c-page__header-brand-link\\">
                <Brand src=\\"logo.png\\" alt=\\"Tekton Hub Logo\\" onClick={[Function: homePage]}>
                  <img onClick={[Function: homePage]} className=\\"pf-c-brand\\" src=\\"logo.png\\" alt=\\"Tekton Hub Logo\\" />
                </Brand>
              </a>
            </div>
            <PageHeaderTools>
              <div className=\\"pf-c-page__header-tools\\">
                <Grid>
                  <div className=\\"pf-l-grid\\" style={[undefined]}>
                    <GridItem span={10}>
                      <div className=\\"pf-l-grid__item pf-m-10-col\\" style={[undefined]}>
                        <Memo(observerComponent)>
                          <TextInput type=\\"search\\" value=\\"\\" onChange={[Function: onSearchChange]} onKeyPress={[Function: onSearchKeyPress]} aria-label=\\"text input example\\" placeholder=\\"Search for resources...\\" spellCheck=\\"false\\" className=\\"hub-search\\">
                            <TextInputBase type=\\"search\\" value=\\"\\" onChange={[Function: onSearchChange]} onKeyPress={[Function: onSearchKeyPress]} aria-label=\\"text input example\\" placeholder=\\"Search for resources...\\" spellCheck=\\"false\\" className=\\"hub-search\\" innerRef={{...}} isRequired={false} validated=\\"default\\" isDisabled={false} isReadOnly={false} isIconSprite={false} isLeftTruncated={false} ouiaSafe={true}>
                              <input onKeyPress={[Function: onSearchKeyPress]} aria-label=\\"text input example\\" placeholder=\\"Search for resources...\\" spellCheck=\\"false\\" onFocus={[Function (anonymous)]} onBlur={[Function (anonymous)]} className=\\"pf-c-form-control hub-search\\" onChange={[Function (anonymous)]} type=\\"search\\" value=\\"\\" aria-invalid={false} required={false} disabled={false} readOnly={false} data-ouia-component-type=\\"PF4/TextInput\\" data-ouia-safe={true} data-ouia-component-id=\\"OUIA-Generated-TextInputBase-1\\" />
                            </TextInputBase>
                          </TextInput>
                        </Memo(observerComponent)>
                      </div>
                    </GridItem>
                    <GridItem span={1} onClick={[Function: onClick]} className=\\"hub-header-search-hint\\">
                      <div className=\\"pf-l-grid__item pf-m-1-col hub-header-search-hint\\" style={[undefined]} onClick={[Function: onClick]}>
                        <Icon id=\\"help\\" size=\\"sm\\" label=\\"search-tips\\">
                          <HelpIcon size=\\"sm\\" label=\\"search-tips\\" color=\\"currentColor\\" noVerticalAlign={false}>
                            <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 1024 1024\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\" label=\\"search-tips\\">
                              <path d=\\"M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C487.9,576 521.3,576 521.3,576 M575.3,751.3 L575.3,655.3 C575.313862,651.055109 573.620137,646.982962 570.6,644 C567.638831,640.947672 563.552355,639.247987 559.3,639.29884 L463.3,639.29884 C459.055109,639.286138 454.982962,640.979863 452,644 C448.947672,646.961169 447.247987,651.047645 447.29884,655.3 L447.29884,751.3 C447.286138,755.544891 448.979863,759.617038 452,762.6 C454.961169,765.652328 459.047645,767.352013 463.3,767.30116 L559.3,767.30116 C563.544891,767.313862 567.617038,765.620137 570.6,762.6 C573.659349,759.643612 575.360354,755.553963 575.3,751.3 M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0\\" />
                            </svg>
                          </HelpIcon>
                        </Icon>
                      </div>
                    </GridItem>
                  </div>
                </Grid>
                <Text style={{...}} component=\\"a\\" onClick={[Function: onClick]}>
                  <a data-ouia-component-type=\\"PF4/Text\\" data-ouia-safe={true} data-ouia-component-id=\\"OUIA-Generated-Text-1\\" style={{...}} onClick={[Function: onClick]} data-pf-content={true} className=\\"\\">
                    <span className=\\"hub-header-login\\">
                      <b>
                        Login
                      </b>
                    </span>
                  </a>
                </Text>
              </div>
            </PageHeaderTools>
          </header>
        </PageHeader>
        <Modal variant=\\"small\\" title=\\"Search tips:\\" isOpen={false} onClose={[Function: onClose]} className=\\"\\" titleIconVariant={{...}} titleLabel=\\"\\" aria-label=\\"\\" showClose={true} aria-describedby=\\"\\" aria-labelledby=\\"\\" id={[undefined]} actions={{...}} hasNoBodyWrapper={false} appendTo={[Function: appendTo]} ouiaSafe={true}>
          <Portal containerInfo={{...}}>
            <ModalContent variant=\\"small\\" isOpen={false} onClose={[Function: onClose]} className=\\"\\" showClose={true} id={[undefined]} actions={{...}} hasNoBodyWrapper={false} boxId=\\"pf-modal-part-0\\" labelId=\\"pf-modal-part-1\\" descriptorId=\\"pf-modal-part-2\\" title=\\"Search tips:\\" titleIconVariant={{...}} titleLabel=\\"\\" aria-label=\\"\\" aria-describedby=\\"\\" aria-labelledby=\\"\\" ouiaId=\\"OUIA-Generated-Modal-small-1\\" ouiaSafe={true} />
          </Portal>
        </Modal>
        <Modal variant=\\"small\\" title=\\"Tekton Hub\\" isOpen={false} onClose={[Function: onClose]} className=\\"hub-header-login__modal\\" aria-label=\\"login\\" titleIconVariant={{...}} titleLabel=\\"\\" showClose={true} aria-describedby=\\"\\" aria-labelledby=\\"\\" id={[undefined]} actions={{...}} hasNoBodyWrapper={false} appendTo={[Function: appendTo]} ouiaSafe={true}>
          <Portal containerInfo={{...}}>
            <ModalContent variant=\\"small\\" isOpen={false} onClose={[Function: onClose]} className=\\"hub-header-login__modal\\" showClose={true} id={[undefined]} actions={{...}} hasNoBodyWrapper={false} boxId=\\"pf-modal-part-1\\" labelId=\\"pf-modal-part-2\\" descriptorId=\\"pf-modal-part-3\\" title=\\"Tekton Hub\\" titleIconVariant={{...}} titleLabel=\\"\\" aria-label=\\"login\\" aria-describedby=\\"\\" aria-labelledby=\\"\\" ouiaId=\\"OUIA-Generated-Modal-small-2\\" ouiaSafe={true} />
          </Portal>
        </Modal>
      </Memo(observerComponent)>
    </Router>
  </BrowserRouter>
   
</Provider>"
`;
